app/flatpak-builtins-install.c

```diff
@@ -300,10 +301,30 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
   if (!flatpak_split_partial_ref_arg (name, &opt_arch, &branch, error))
     return FALSE;
 
+  /* Make sure the default branch is pulled from the server's summary
+     file if it's defined there and hasn't been pulled in yet. */
+  default_branch = flatpak_dir_get_remote_default_branch (dir, repository);
```

Comment:
> [...]
> But at that point I think I'd rather remove the fetching part from here and provide a new built-in command that would update the local information we have for the configured remotes with the one from the server's summary file (i.g. `flatpak remote-update`?), expose it via the public API and then maybe call it from GNOME Software on startup?

JFTR, I mentioned this on IRC and we agreed having this functionality refactored somewhere else and exposed via a new `remote-update` command and new public API would be a more interesting approach, so I did that and then removed these changes from the `install` command.

The new relevant commits, pushed on top of this PR, are:
  * [New internal API to update configuration of remotes from their summary file](https://github.com/flatpak/flatpak/pull/345/commits/98fdd2651b5f32b5c5312bad43dceededf2801ad)
  * [Added new flatpak built-in command: remote-update](https://github.com/flatpak/flatpak/pull/345/commits/47cc56fe0ff02225b9d91eeb67caed0aa8cb705a)
  * [Added documantation for the new flatpak remote-update comand](https://github.com/flatpak/flatpak/pull/345/commits/3e8e5671df4838da16398f75f2cb9067f16fec60)
  * [Added new public API: flatpak_installation_update_remote_sync()](https://github.com/flatpak/flatpak/pull/345/commits/d451c2090dd5de55fa771bc6cf8a608b429e73c4)

I think this should be all we would need for our use case with GNOME Software: instead of automatically updating the local summary file on every install, we'd now decide what to do from GNOME Software itself, which kind of makes sense in my mind.